execution/src/canonical.rs

```diff
@@ -201,6 +208,16 @@ pub(super) fn execute_canonical_transaction<
         }
     };
 
+    let vm_outcome = vm_outcome.and_then(|_| {
+        // Ensure any base token balance changes in EVM are reflected in Move too
+        eth_token::replicate_transfers(
```

Comment:
AFAICT transfers are replicated after getting the VM outcome for both the canonical and deposited tx flow. Any way we could abstract it to a higher level so that it's e.g. called in e.g. `execute_transaction()`? Or does it break the transfer logic?

Reply:
The reason we need to do it here is because it needs to have the session object still in scope. The session is what ultimately creates the MoveVM changes that we return from this function. We could refactor to return the session object up to `execute_transaction`, but I don't think that will make things cleaner because there is different post-processing in the case of canonical vs deposit.

Reply:
I see, thanks for clarifying!